tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -169,6 +179,16 @@ private int Serialize(ref byte[] bytes, int offset, in SpanModel spanModel)
                 len++;
             }
 
+            var hasSpanEvents = span.SpanEvents is { Count: > 0 };
+            var nativeSpanEventsEnabled = (span.Context.TraceContext?.Tracer as Tracer)?.TracerManager?.SpanEventsManager?.NativeSpanEventsEnabled;
```

Comment:
When we need a value for serialization, we've been adding them to `SpanModel` or `TraceChunkModel`, like here: https://github.com/DataDog/dd-trace-dotnet/blob/b2f1c6464cf6212d413eaf449efc7e6784412ade/tracer/src/Datadog.Trace/Agent/MessagePack/TraceChunkModel.cs#L32C1-L58C46

The idea is to move away from accessing the `Tracer` as much as possible to make it easier to test `SpanMessagePackFormatter` without having to set everything up (tracer, settings, etc).